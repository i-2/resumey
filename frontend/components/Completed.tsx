import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Button, Flex, Heading, Text } from '@chakra-ui/react';
import * as actionCreators from '../actions';

export const _Completed = (props: any) => {
    const { hasPublished, urlLink } = props.publish;
    const onPublish = () => props.tryPublish();
    return (
        <Flex width="100%" direction="column" justify="center" borderWidth="1px" borderRadius="lg" padding="5%" marginTop="2%">
            <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%"><Heading>  Almost Complete! </Heading> </Flex>
            {
                !hasPublished ? <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%"><Button colorScheme="teal" onClick={onPublish}> Publish Now </Button></Flex> :
                    <Flex width="100%" direction="row" justify="center" padding="5%" marginTop="2%">
                        <a href={urlLink}>
                            <Text>{urlLink}</Text>
                        </a>
                    </Flex>
            }
        </Flex>
    )
}

const mapStateToProps = (state: any) => ({
    publish: state.publish
})
const mapDispatchToProps = (dispatch: any) => bindActionCreators(actionCreators, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(_Completed);