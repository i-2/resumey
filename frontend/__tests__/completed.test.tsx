import * as  React from 'react';
import * as renderer from 'react-test-renderer';
import Completed from '../components/Completed';

it('renders completed correctly', () => {
  const tree = renderer
    .create(<Completed />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});