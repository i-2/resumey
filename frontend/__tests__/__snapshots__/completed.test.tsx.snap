// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders completed correctly 1`] = `
<p
  className="chakra-text css-0"
>
   Completed !! 
</p>
`;
